### PR TITLE
Suggestion: Add filenames to Refine method

### PR DESCRIPTION
**Pull request by FergusFettes**
I found the refine method when you add the filenames as part of the context. Often the filenames provide very useful information about the file in question, and help the agent get an idea for the structure of the repo you are summarizing (or whatever).

If you like this suggestion I can add more details about how I use it, something like:
```
text_docs = [Document(d, extra_info={"filepath": str(f.absolute())}) for d, f in zip(data_list, self.input_files)]
```

### CODE REVIEW

File: langchain/chains/combine_documents/refine.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 
 def _get_default_document_prompt() -> PromptTemplate:
-    return PromptTemplate(input_variables=["page_content"], template="{page_content}")
+    return PromptTemplate(input_variables=["page_content", "file_path"], template="File:\n{file_path}\nContent:\n{page_content}")
```
Comment (vowelparrot) [inline langchain/chains/combine_documents/refine.py:20]: I think the idea of including source data in the document is a good one - the Documents don't necessarily come from "files" (they could come from anywhere) so we'll have to consider what the best default would be for this

Comment (dev2049) [reply langchain/chains/combine_documents/refine.py:20]: yea not convinced this is generic enough to be set as default prompt, users can always override default if they'd like

Comment (FergusFettes) [reply langchain/chains/combine_documents/refine.py:20]: yeah maybe its more about having some extra default info (if available)

but feel free to close this if you think there is a better way to do it

**Comment by hwchase17** [comment]
yeah i dont think general enough to be default, but would be worth having an example to pass in!

**Pull request closed without merging**